src/weather/mod.rs

```diff
@@ -19,31 +20,40 @@ pub trait WeatherProvider {
 
 #[derive(Clone, Debug)]
 pub enum Provider {
+    // contains api key
     WeatherAPI(String),
     OpenMeteo,
 }
 
 impl CurrentConditions {
     pub fn get(
-        provider: Provider,
+        providers: Vec<Provider>,
         unit: crate::Unit,
         latitude: &str,
         longitude: &str,
     ) -> eyre::Result<CurrentConditions> {
-        let client: Box<dyn WeatherProvider> = match provider {
-            Provider::WeatherAPI(key) => Box::new(weather_api::Client::new(
-                key,
-                latitude.to_string(),
-                longitude.to_string(),
-            )),
-            Provider::OpenMeteo => Box::new(open_meteo::Client::new(
-                unit,
-                latitude.to_string(),
-                longitude.to_string(),
-            )),
-        };
-
-        client.current()
+        for provider in providers {
```

Comment:
Should be able to setup a test that passes a mock-http client that will fail for weatherapi so that we can ensure fallback to open-meteo.